src/D2L.Bmx/Okta/OktaApi.cs

```diff
@@ -73,17 +78,19 @@ internal class OktaApi : IOktaApi {
 				authnResponse.Embedded.Factors
 			);
 		}
-		throw new BmxException( "Error authenticating to Okta" );
+		throw new BmxException( "Okta authentication failed. Check if org, user and password is correct" );
 	}
 
 	async Task IOktaApi.IssueMfaChallengeAsync( string stateToken, string factorId ) {
-		var response = await _httpClient.PostAsJsonAsync(
+		try {
+			var response = await _httpClient.PostAsJsonAsync(
 			$"authn/factors/{factorId}/verify",
 			new IssueMfaChallengeRequest( stateToken ),
 			SourceGenerationContext.Default.IssueMfaChallengeRequest );
 
-		if( !response.IsSuccessStatusCode ) {
-			throw new BmxException( "Error sending MFA challenge" );
+			response.EnsureSuccessStatusCode();
+		} catch( Exception ex ) {
+			throw new BmxException( "Error sending MFA challenge", ex );
```

Comment:
I know you only kept the original message here, but I'm thinking we can reword this to be more user friendly. e.g.

> Error initiating MFA with Okta

maybe? is this an accurate description?